src/serialize.h

```diff
@@ -148,7 +148,14 @@ enum
     SER_GETHASH         = (1 << 2),
 };
 
-#define READWRITE(...)      (::SerReadWriteMany(s, ser_action, __VA_ARGS__))
+//! Convert the reference base type to X, without changing constness or reference type.
+template<typename X> X& ReadWriteAsHelper(X& x) { return x; }
+template<typename X> const X& ReadWriteAsHelper(const X& x) { return x; }
+template<typename X> X&& ReadWriteAsHelper(X&& x) { return std::move(x); }
```

Comment:
As `x` is `X&&`,  is `std::move` necessary?

Reply:
`x` is a variable, and thus can't be a temporary.

The `X&&` only makes that function call match the case where a temporary is passed. Inside that function it's just a normal variable.

Reply:
@sipa thanks,  found a verbose explanation [here](https://stackoverflow.com/a/7518365).   Had been mislead by a previous generalization of a possible `std::move` implementation that lacked the cast.